pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -444,13 +443,13 @@ func (r *controlPlaneInstanceReconciler) renderCharts(ctx context.Context) error
 	r.Status.LastAppliedConfiguration = r.Instance.Spec
 	if len(r.Status.LastAppliedConfiguration.Version) == 0 {
 		// initialize version
-		if len(r.Status.AppliedVersion) == 0 {
+		if len(r.Instance.Spec.Version) == 0 {
 			// this must be from a 1.0 operator
 			r.Status.LastAppliedConfiguration.Version = maistra.LegacyVersion.String()
 		} else {
 			// use the version that's already been applied
 			// users must manually upgrade their mesh version
-			r.Status.LastAppliedConfiguration.Version = r.Status.AppliedVersion
+			r.Status.LastAppliedConfiguration.Version = r.Instance.Spec.Version
```

Comment:
I think this can be simplified. This `else` block here seems like it can never be hit anyway

We're writing `r.Instance.Spec` into `LastAppliedConfiguration` above in line 443, so `LastAppliedConfiguration.Version` will always be equal to `r.Instance.Spec.Version`. I think this inner `if-else` block can be removed

Reply:
Good catch. Fixed.

Reply:
Except for the case where `Version == ""`, in which case it gets set to `v1.0`.

Reply:
@rcernich Yup, I kept that part.